### PR TITLE
Comparision of velocity dispersion distribution from skypy and SL-Hammocks pipeline

### PR DESCRIPTION
**Pull request by nkhadka21**
This PR fixes a bug in some velocity dispersion functions and ensures that the velocity dispersion distribution from skypy  and SL-Hammock pipelines match with each other.

### CODE REVIEW

File: slsim/Deflectors/velocity_dispersion.py [hunk 1]
```diff
@@ -415,16 +427,71 @@ def f(lnx, a):
...
-        redshift=redshift,
-        density=density,
-        sky_area=sky_area,
-        cosmology=cosmology,
-        noise=noise,
+        redshift, density, sky_area, cosmo, noise=noise
     )
 
 
+def redshifts_from_comoving_density(redshift, density, sky_area, cosmo, noise=True):
```
Comment (sibirrer) [inline slsim/Deflectors/velocity_dispersion.py:434]: is this definition from SkyPy? If so, please acknowledge it properly

File: slsim/Deflectors/velocity_dispersion.py [hunk 2]
```diff
@@ -472,16 +541,33 @@ def schechter_velocity_dispersion_function(
     .. [3] Choi, Park and Vogeley, (2007), astro-ph/0611607, doi:10.1086/511060
     """
 
-    if np.ndim(alpha) > 0:
-        raise NotImplementedError("only scalar alpha is supported")
+    if size is None:
+        size = np.broadcast(vd_min, vd_max, scale).shape or None
```
Comment (sibirrer) [inline slsim/Deflectors/velocity_dispersion.py:545]: this line/possibility is not tested

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you very much @nkhadka21! I have two minor points and then it's ready to merge

File: slsim/Deflectors/velocity_dispersion.py [hunk 3]
```diff
@@ -228,10 +226,10 @@ def vel_disp_sdss(sky_area, redshift, vd_min, vd_max, cosmology, noise=True):
     .. [1] Choi, Park and Vogeley, (2007), astro-ph/0611607, doi:10.1086/511060
     """
     # SDSS velocity dispersion function for galaxies brighter than Mr >= -16.8
-    phi_star = 8.0 * 10 ** (-3) / cosmology.h**3
-    vd_star = 161
-    alpha = 2.32
-    beta = 2.67
+    phi_star = 2.099e-2 * cosmology.h**3
```
Comment (sibirrer) [inline slsim/Deflectors/velocity_dispersion.py:229]: just checking, are these the numbers on Choi et al. 2007 or are these from a different paper?

**Review by sibirrer — COMMENTED** [review]
I have one comment to confirm, otherwise good to go

File: slsim/Deflectors/velocity_dispersion.py (hunk 3 above)
Comment (nkhadka21) [reply slsim/Deflectors/velocity_dispersion.py:231]: Ohh, these numbers are from Bernardi et al. 2010. I updated the reference.

**Review by sibirrer — APPROVED** [review]

**Pull request merged**